tests/test_problem_bank_helpers_rounding.py

```diff
@@ -0,0 +1,221 @@
+    (3.14159, 2, '3.14'),  # Test rounding a positive float to 2 digits after decimal
+    (-2.71828, 2, '-2.72'),  # Test rounding a negative float to 2 digits after decimal
+    (0, 2, '0.00'),  # Test rounding zero to 2 digits after decimal
+    (123.456, 0, '123'),  # Test rounding a float to 0 digits after decimal
+    (7.7777777, 2, '7.78'),  # Test rounding a float with more than 2 digits after decimal
+    (9.99, 2, '9.99'),  # Test rounding a float with exactly 2 digits after decimal
+    (2.555, 4, '2.5550'),  # Test rounding a float with fewer digits after decimal than specified
+    (1.23456789, 6, '1.234568'),  # Test rounding a float with more digits after decimal than specified
+    (1e18, 2, '1000000000000000000.00'),  # Test rounding a very large positive float
+    (1e-18, 2, '0.00'),  # Test rounding a very small positive float (close to zero)
+    (-1e18, 2, '-1000000000000000000.00'),  # Test rounding a very large negative float
+    (-1e-18, 2, '-0.00'),  # Test rounding a very small negative float (close to zero)
+    (1.999, 2, '2.00'),  # Test rounding up
+    (2.001, 2, '2.00'),  # Test rounding down
+    (1.555, 2, '1.56'),  # Test rounding halfway between two rounded values
+    (-1.5, 2, '-2.00'),  # Test rounding halfway between negative and positive values
+    (1234567890.123456789, 8, '1234567890.12345679'),  # Test rounding a float with a large number of digits before and after decimal
+    (987.654321, -2, '1000.00'),  # Test rounding a float with a negative number of digits after decimal
+    (42.123, 0, '42'),  # Test rounding a float with zero value for digits_after_decimal
+    (True, 2, '1.00'),  # Test rounding a non-numeric input
+    (10.00, 2, '10.00'),  # Test rounding a float that is already rounded to specified digits after decimal
+])
+def test_num_as_str(num, digits_after_decimal, expected_result):
+    assert pbh.num_as_str(num, digits_after_decimal) == expected_result
+
+
+def test_num_as_str_invalid_args_kwargs():
+    with pytest.raises(TypeError):
+        pbh.num_as_str(123.45, 2, "args")  # Function should not accept *args
+
+    with pytest.raises(TypeError):
+        pbh.num_as_str(123.45, 2, digits_after_decimal=2)  # Function should not accept **kwargs
+
+
+# Test roundp function
+def test_roundp():
+    """Test rounding a float with specified sigfigs"""
+    assert pbh.roundp(123.456, sigfigs=2) == 120.0
+
+    """Test rounding an integer with specified decimals"""
+    assert pbh.roundp(987, decimals=2) == 987.0
+
+    """Test rounding a string with default sigfigs"""
+    assert pbh.roundp('99.8765') == '100'
+
+    """Test rounding a float with format='std'"""
+    assert pbh.roundp(0.123, format='std') == 0.12
+
+    """Test rounding a float with format='English'"""
+    assert pbh.roundp(12345.6789, format='English') == 12300.0
+
+    """Test rounding a float with format='sci'"""
+    assert pbh.roundp(0.000123, format='sci') == '1.20e-04'
```

Comment:
```suggestion
def test_roundp_with_overriden_sigfig_settings():
    """Test rounding a float with specified sigfigs"""
    assert pbh.roundp(123.456, sigfigs=2) == 120.0

def test_roundp_with_overriden_decimal_settings():
    """Test rounding an integer with specified decimals"""
    assert pbh.roundp(987, decimals=2) == 987.0

def test_roundp_with_default_settings():
    """Test rounding a string with default sigfigs"""
    assert pbh.roundp('99.8765') == '100'

def test_roundp_with_format_std():
    """Test rounding a float with format='std'"""
    assert pbh.roundp(0.123, format='std') == 0.12

def test_roundp_with_format_english():
    """Test rounding a float with format='English'"""
    assert pbh.roundp(12345.6789, format='English') == 12300.0

def test_roundp_format_sci():
    """Test rounding a float with format='sci'"""
    assert pbh.roundp(0.000123, format='sci') == '1.20e-04'
```
Suggestion: Separate these into separate tests, so if the function ever changes or regresses from expected state, we can see multiple failures in a single test run.